AndroidX

forkch opened the pull request:
migration to AndroidX

right now i cannot build due to some maven publishing taks which are not working on my machine (or due to the new gradle plugin..)

shiraji commented on processor/build.gradle:
```diff
@@ -36,8 +36,8 @@ dependencies {
-    testCompileAar "com.android.support:support-v4:$SUPPORT_LIBRARY_VERSION"
-    testCompileAar "com.android.support:support-v13:$SUPPORT_LIBRARY_VERSION"
+    testCompileAar 'androidx.legacy:legacy-support-v4:1.0.0-alpha3'
+    testCompileAar 'androidx.legacy:legacy-support-v13:1.0.0-alpha3'
```
Create a version code at `gradle.properties`

shiraji replied:
TO: @hotchemi @mannodermaus 

I was thinking we are still keep android support library's support. Do we want to drop the support of support library?

shiraji commented on sample-kotlin/src/main/kotlin/permissions/dispatcher/samplekotlin/camera/CameraPreviewFragment.kt:
```diff
@@ -26,7 +24,7 @@ import permissions.dispatcher.samplekotlin.R
  * http://developer.android.com/guide/topics/media/camera.html
  */
-class CameraPreviewFragment : Fragment() {
+class CameraPreviewFragment : androidx.fragment.app.Fragment() {
```
Why do not `import`?

forkch commented:
I've fixed the mentioned issues (imports and versioning) and fixed the travis build by copying the licences to $ANDROID_HOME/licences

It is failing now due to the publishing plugin. I don't know how you usually deal with this.

hotchemi commented:
Might be related to https://github.com/novoda/bintray-release could you try to update?

forkch commented:
no luck...bintray-release does not seem to support gradle-4.6 but the android-gradle plugin for AndroidX should be the 3.2.0-alpha* 

https://github.com/novoda/bintray-release/issues/177

hotchemi commented:
Umm.. 🤔 
I found kind of official plugin and seems there is a PR for that but haven't merged yet.
https://github.com/bintray/gradle-bintray-plugin/pull/230

forkch commented:
Unfortunately I have no resources atm to look into bintray (have not yet deployed any artefacts to any kind of artifactory :-)).

hotchemi commented:
Umm ok, it would be actually a huge change so it might be better to be addressed by @permissions-dispatcher/maintainers 🤔Really sorry to say that but this is actually a great work!

shiraji commented:
Let me take over this work.

shiraji commented:
@forkch if you don't mind, could I change your author mail address to your GitHub email address? This change lead to associate your commit with your GitHub account.

forkch commented:
Sure no problem!

shiraji commented:
Done. See #494

I will close this PR since all @forkch works are there. Thank you for your contribution!

hotchemi commented:
Thank you @forkch !
